lib/src/index.tsx

```diff
@@ -197,7 +197,8 @@ function parseChildren(
     }
   } else if (typeof children === "object") {
     const subProps = React.Children.toArray(children as any).map((c: any) => {
-      const codeProps = c.props?.children?.props
+      const codeProps = c.props?.children?.props ?? c.props
```

Comment:
I was originally concerned whether this would have adverse effects, but after getting to know the code, I think `c.props` is actually what we want in this case; It's fundamentally the same situation as the `if` block on Line 193, but the `type` of the element is a function instead of the string "code".

This effectively ensures we're doing the same thing either way.